internal/experiments/experiments_remote_execute_api.go

```diff
@@ -0,0 +1,205 @@
+	ServiceAccountName   string `yaml:"service_account_name"`
+	LocalPort            int32  `yaml:"local_port"`
+	ExpectedHTTPResponse int    `yaml:"expected_response"`
+	ExpectedBody         string `yaml:"expected_body"`
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Type() string {
+	return "remote_execute_api"
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Description() string {
+	return "This experiment runs a deployment based on a configurable image and then verifies based off of API calls to that image"
+}
+func (p *RemoteExecuteAPIExperimentConfig) Technique() string {
+	return categories.MITRE.Execution.NewContainer.Technique
+}
+func (p *RemoteExecuteAPIExperimentConfig) Tactic() string {
+	return categories.MITRE.Execution.NewContainer.Tactic
+}
+func (p *RemoteExecuteAPIExperimentConfig) Framework() string {
+	return string(categories.Mitre)
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Run(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) error {
+	var config RemoteExecuteAPIExperimentConfig
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return err
+	}
+
+	executorConfig := executor.NewExecutorConfig(
+		config.Metadata.Name,
+		config.Metadata.Namespace,
+		config.Parameters.Image,
+		config.Parameters.ImageParameters,
+		config.Parameters.ServiceAccountName,
+		config.Parameters.TargetPort,
+		config.Parameters.LocalPort,
+	)
+
+	err = executorConfig.Deploy(ctx, client.Clientset)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Verify(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) (*verifier.Outcome, error) {
+	var config RemoteExecuteAPIExperimentConfig
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	executorConfig := executor.NewExecutorConfig(
+		config.Metadata.Name,
+		config.Metadata.Namespace,
+		config.Parameters.Image,
+		config.Parameters.ImageParameters,
+		config.Parameters.ServiceAccountName,
+		config.Parameters.TargetPort,
+		config.Parameters.LocalPort,
+	)
+
+	stopChan := make(chan struct{}, 1)
+	readyChan := make(chan struct{}, 1)
+	errch := make(chan error, 1)
+	out, errOut := new(bytes.Buffer), new(bytes.Buffer)
+
+	forwardConfig := executor.PortForwardConfig{
+		StopCh:  stopChan,
+		ReadyCh: readyChan,
+		Out:     out,
+		ErrOut:  errOut,
+	}
+
+	go func() {
+		err := executorConfig.OpenLocalPort(ctx, client, config.Parameters.LocalPort, forwardConfig)
+		if err != nil {
+			errch <- fmt.Errorf("Local Port Failed to open: %w", err)
+		}
+	}()
+
+	// Waits until local port is ready or open erros
+	select {
+	case <-readyChan:
+		break
+	case err := <-errch:
+		return nil, err
+	}
+
+	path := config.Parameters.Path
+	port := config.Parameters.LocalPort
+	url := url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s:%d", "127.0.0.1", port),
```

Comment:
If the `127.0.0.1` is always a constant due to the port forwarding, why not do `executor.Addr` and define a const in the executor package?